golang/cosmos/x/vstorage/keeper/keeper.go

```diff
@@ -156,14 +192,23 @@ func (k Keeper) EmitChange(ctx sdk.Context, change *ProposedChange) {
 	)
 }
 
-// GetData gets generic storage.  The default value is an empty string.
-func (k Keeper) GetData(ctx sdk.Context, path string) string {
-	//fmt.Printf("GetData(%s)\n", path);
+// GetEntry gets generic storage.  The default value is an empty string.
+func (k Keeper) GetEntry(ctx sdk.Context, path string) types.StorageEntry {
+	//fmt.Printf("GetEntry(%s)\n", path);
 	store := ctx.KVStore(k.storeKey)
 	encodedKey := types.PathToEncodedKey(path)
-	bz := bytes.TrimPrefix(store.Get(encodedKey), types.EncodedDataPrefix)
-	value := string(bz)
-	return value
+	rawValue := store.Get(encodedKey)
+	if len(rawValue) == 0 {
+		return types.NewStorageEntryWithNoData(path)
+	}
+	if bytes.Equal(rawValue, types.EncodedNoDataValue) {
+		return types.NewStorageEntryWithNoData(path)
+	}
+	if !bytes.HasPrefix(rawValue, types.EncodedDataPrefix) {
+		panic(fmt.Errorf("value at path %q starts with unexpected prefix", path))
+	}
+	value := string(rawValue[len(types.EncodedDataPrefix):])
```

Comment:
```suggestion
	// TODO: Use bytes.CutPrefix once we can rely upon go >= 1.20.
	// value, hasPrefix := bytes.CutPrefix(rawValue, types.EncodedDataPrefix)
	// if !hasPrefix {
	// 	panic(fmt.Errorf("value at path %q starts with unexpected prefix", path))
	// }
	if !bytes.HasPrefix(rawValue, types.EncodedDataPrefix) {
		panic(fmt.Errorf("value at path %q starts with unexpected prefix", path))
	}
	value := string(rawValue[len(types.EncodedDataPrefix):])
```
And given the reuse, it might even make sense to just define a file-local `func cutPrefix(s, prefix []byte) (after []byte, found bool)`.
```suggestion
	value, hasPrefix := cutPrefix(rawValue, types.EncodedDataPrefix)
	if !hasPrefix {
		panic(fmt.Errorf("value at path %q starts with unexpected prefix", path))
	}
```